src/main/java/pl/allegro/tech/discovery/consul/recipes/watch/ConsulLongPollCallback.java

```diff
@@ -193,10 +192,17 @@ private long updateIndex(Response response) {
         }
 
         long index = Long.valueOf(indexString);
-        if (currentIndex.get() == index) {
+        long lastIndex = currentIndex.get();
+
+        if (index == lastIndex) {
             stats.indexNotChanged();
             logger.trace("Discarding event on endpoint {} index {} as index did not change", endpoint, index);
             return -1;
+        } else if (index < lastIndex) {
+            stats.indexBackwards();
+            logger.warn("Discarding event on endpoint {} as new index index {} is lower than previous {} - resetting index", endpoint, index, lastIndex);
+            currentIndex.set(0);
+            return -1;
```

Comment:
We should return 0 here.

Reply:
why?  (see comment above)

Reply:
ok, switched to enum.  seems more readable indeed.